website/integrations/sources/discord/index.md

```diff
@@ -154,3 +154,205 @@ return user_matched
+groups_to_add = []
+for role_id in MAPPED_ROLES:
+    if role_id in guild_member_info["roles"]:
+        groups_to_add.append(MAPPED_ROLES[role_id])
+
+request.context["flow_plan"].context["groups"] = groups_to_add
+return True
+
+```
+
+Now bind this policy to the chosen enrollment flows for the Discord OAuth source.
+
+#### Sync on authentication
+
+```python
+# To get the role and guild ID numbers for the parameters, open Discord, go to Settings > Advanced and
+# enable developer mode.
+# Right-click on the server/guild title and select "Copy ID" to get the guild ID.
+# Right-click on the server/guild title and select server settings > roles, right click on the role and click
+# "Copy ID" to get the role ID.
+
+from authentik.core.models import Group
+
+GUILD_ID = "<YOUR GUILD ID>"
+MAPPED_ROLES = {
+    "<Discord Role Id>": Group.objects.get_or_create(name="<Authentik Role Name>")[0],
+}
+
+# Only change below here if you know what you are doing.
+GUILD_API_URL = "https://discord.com/api/users/@me/guilds/{guild_id}/member"
+
+# Ensure flow is only run during OAuth logins via Discord
+if context["source"].provider_type != "discord":
+    return True
+
+# Get the user-source connection object from the context, and get the access token
+connection = context.get("goauthentik.io/sources/connection")
+if not connection:
+    return False
+access_token = connection.access_token
+
+guild_member_info = requests.get(
+    GUILD_API_URL.format(guild_id=GUILD_ID),
+    headers={"Authorization": "Bearer " + access_token},
+).json()
+
+# Ensure user is a member of the guild
+if "code" in guild_member_info:
+    if guild_member_info["code"] == 10004:
+        ak_message("User is not a member of the guild")
+    else:
+        ak_create_event(
+            "discord_error", source=context["source"], code=guild_member_info["code"]
+        )
+        ak_message("Discord API error, try again later.")
+    return False
+
+# Get the user's current roles and remove all roles we want to remap
+new_groups = [
+    role for role in request.user.ak_groups.all() if role not in MAPPED_ROLES.values()
+]
+
+# Add back mapped roles which the user has in the guild
+for role_id in MAPPED_ROLES:
+    if role_id in guild_member_info["roles"]:
+        new_groups.append(MAPPED_ROLES[role_id])
+
+# Update user's groups
+request.user.ak_groups.set(new_groups)
+request.user.save()
+
+return True
+
+```
+
+Now bind this policy to the chosen authentication flows for the Discord OAuth source.
+
+### Store oauth info in attribute and create avatar attribute from discord avatar
+
+:::info
+Ensure that the Discord OAuth source in 'Federation & Social login' has the additional `guilds.members.read` scopes added under the 'Protocol settings'.
+:::
+
+:::info
+In order to use the created attribute in authentik you will have to set authentik configuration arguments found at: https://goauthentik.io/docs/installation/configuration#authentik_avatars
```

Comment:
They might have moved the location of the avatars here:
https://docs.goauthentik.io/docs/core/settings#avatars

Reply:
You're right. I'll fix that :)